packages/preview/sweet-graduate-resume/0.1.0/template.typ

```diff
@@ -0,0 +1,139 @@
+#import "lib_fontawesome.typ": *
+
+#let preamble(doc) = {
+  set page(
+    paper: "a4",
+    margin: (x: 0.5cm, y: 0.5cm),
+  )
+
+  set text(
+    font: "DejaVu Sans",
+    size: 8pt,
+  )
+
+  set par(justify: true)
+
+  show link: name => {
+    set text(blue)
+    name
+  }
+
+  set block(spacing: 0.5em)
+
+  set align(left)
+  doc
+}
+
+#let header(author, roll, school, urls) = {
+  set text(17pt, weight: "bold")
+  set block(spacing: 0.3em)
+  [
+    #author | #roll
+
+    #set text(13pt)
+    #box(
+      width: 1fr,
+      inset: (bottom: 5pt),
+      stroke: (bottom: (1pt + black)),
+    )[#school]
+
+  ]
+
+  set text(7pt)
+  for value in urls {
+    let img = if value.fa {
+      if value.brand {
+        fa-icon(value.svg)
+      } else {
+        fa-icon(value.svg, solid: value.solid)
+      }
+    } else {
+      image("template/svg/" + value.svg)
```

Comment:
Unfortunately, this will not work (sorry, I could probably have noticed that in the last review too). `svg` would have to be passed as an `image` element directly, and not a string giving the filename. This is because packages can't access files outside of their folder (the template folder would be copied to the user's project, it is not considered to be part of the package, and I think you want to allow people to add their own SVG files, not just use the two Codeberg logos you provide).

So this line should become : `value.svg`, and the elements of `urls` should look like that:

```typ
#let urls = (
  (
    name: "Codeberg",
    url: "https://codeberg.org/innocent_zero",
    svg: image("svg/codeberg.svg"), // Here is the change
    fa: false,
    brand: false,
    solid: false,
  ),
  (name: "Website", url: "https://example.com", svg: "link", fa: true, brand: false, solid: false),
  (name: "Website", url: "https://example.com", svg: "github", fa: true, brand: true, solid: true),
)
```

Reply:
Hi, really sorry for not checking it. I simply checked if typst was able to compile it or not without caring about the pdf. I've fixed it now. Thanks for your help!